src/scheduler/fifo.c

```diff
@@ -2608,7 +2608,10 @@ update_svr_schedobj(int connector, int cmd, int alarm_time)
 		return 1;
 
 	/* Stat the scheduler to get details of sched */
-	ss = pbs_statsched(connector, sc_name, NULL, NULL);
+	ss = pbs_statsched(connector, NULL, NULL);
+	while(ss && strncmp(sc_name, ss->name, PBS_MAXSCHEDNAME)) {
```

Comment:
Check ss != NULL.  Also, I don't think you do an strncmp().  A strcmp() will be fine.  Lastly, you are iterating, use a for() loop.

Reply:
1.  `while (ss)` == `while (ss != NULL)` . I prefer former succinct format.
2.  why let `strcmp()` run amok if by chance we get a non-null termintated string. I say its safe to restrict it to "n"
3.  Ok I will change that, but I dont think there will be any difference in the compiled binary code

Reply:
1. Follow scheduler style: use var != NULL.
2. The problem here is you're assuming the values will be PBS_MAXSCHEDNAME.  While this should be the case, the IFL interface doesn't limit the names it returns.  In the case where you have one string longer than the other with the same initial characters, you'll pass this test.  You should be in no danger of having non-null terminated strings.  IFL will terminate its strings.
3. A for() loop is a safer construct for iterating than a while() loop.  If at any point you use a continue in a while loop and forget to move your iteration variable along,  you'll get into an infinite loop.  In any case, follow scheduler style.  Use a for() loop.